### PR TITLE
ci: run ci on pull requests and pushes to master

### PR DESCRIPTION
**Pull request by aminya**
This makes it easier to test what a PR is adding to the project.

### CODE REVIEW

**Review by ddalcino — APPROVED** [review]
Yes, this is sorely needed! Thanks!

**Comment by lefticus** [comment]
I'm in the process of working on the conan issues that are currently preventing builds from running. I'll approve this PR and get a new one up shortly.

File: .github/workflows/build_cmake.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: CMake
 
-on: [push]
+on:
+  pull_request:
+  push:
+    branches:
+      - master
```
Comment (lefticus) [inline .github/workflows/build_cmake.yml:7]: Should this be `main` now?

Comment (aminya) [reply .github/workflows/build_cmake.yml:7]: Oh, right. 
```suggestion
      - main
```

**Review by lefticus — APPROVED** [review]

**Pull request merged**